src/main/java/dook/services/Parser.java

```diff
@@ -0,0 +1,177 @@
+        return new AddTaskCommand(task);
+    }
+    private Command handleDeadline(String body) throws DookException {
+        if (body.isBlank()) {
+            throw new DookException("Usage: deadline [name] /by [time].");
+        }
+
+        String[] tmp = body.split("/by", 2);
+        if (tmp.length <= 1) {
+            throw new DookException("Usage: deadline [name] /by [time].");
+        }
+
+        String desc = tmp[0].trim();
+        String by = tmp[1].trim();
+        if (desc.isBlank() || by.isBlank()) {
+            throw new DookException("Some information is missing!\n"
+                    + "Usage: deadline [name] /by [time].");
+        }
+        Task task = new Deadline(desc, by, false);
+        return new AddTaskCommand(task);
+    }
+    private Command handleEvent(String body) throws DookException {
+        if (body.isBlank()) {
+            throw new DookException("Usage: event [name] /from [start] /to [end].");
+        }
+        String[] tmp1 = body.split("/from", 2);
+        if (tmp1.length <= 1) {
+            throw new DookException("Usage: event [name] /from [start] /to [end].");
+        }
+
+        String desc = tmp1[0].trim();
+
+        String[] tmp2 = tmp1[1].split("/to", 2);
+        if (tmp2.length <= 1) {
+            throw new DookException("Usage: event [name] /from [start] /to [end].");
+        }
+        String from = tmp2[0].trim();
+        String to = tmp2[1].trim();
+        if (desc.isBlank() || from.isBlank() || to.isBlank()) {
+            throw new DookException("Some information is missing!\n"
+                    + "Usage: event [name] /from [start] /to [end].");
+        }
+
+        Task task = new Event(desc, from, to, false);
+        return new AddTaskCommand(task);
+    }
+    private Command handleDelete(String body) throws DookException {
+        int index;
+        try {
+            index = Integer.parseInt(body.split(" ", 2)[0]);
+        } catch (NumberFormatException e) {
+            throw new DookException("Usage: delete [task number]");
+        }
+        return new DeleteTaskCommand(index);
+    }
+    private Command handleBefore(String body) throws DookException {
+        LocalDate localDate;
+        try {
+            localDate = TimeProcessor.getLocalDateFromString(body.split(" ", 2)[0].trim());
+        } catch (DateTimeParseException e) {
+            throw new DookException("Improper date format.");
+        }
+        return new BeforeCommand(localDate);
+    }
+
+    private Command handleAfter(String body) throws DookException {
+        LocalDate localDate;
+        try {
+            localDate = TimeProcessor.getLocalDateFromString(body.split(" ", 2)[0].trim());
+        } catch (DateTimeParseException e) {
+            throw new DookException("Improper date format.");
+        }
+        return new AfterCommand(localDate);
+    }
+
+    private Command handleDuring(String body) throws DookException {
```

Comment:
I like how you name these methods with the same prefix to handle different cases.